lib/multigrid.cpp

```diff
@@ -780,13 +761,14 @@ namespace quda
 
     // temporary fields used for verification
     std::vector<ColorSpinorField> fine_tmp(param.Nvec);
-    ColorSpinorParam fine_param(r[0]);
-    fine_param.create = QUDA_NULL_FIELD_CREATE;
+    ColorSpinorParam fine_param(param.B[0]);
+    fine_param.setPrecision(param.mg_global.invert_param->cuda_prec_sloppy, QUDA_INVALID_PRECISION,
+                            fine_param.location == QUDA_CUDA_FIELD_LOCATION ? true : false);
+    fine_param.gammaBasis = param.level > 0 ? QUDA_DEGRAND_ROSSI_GAMMA_BASIS : QUDA_UKQCD_GAMMA_BASIS;
```

Comment:
This line breaks staggered --- if `param.B[0].Nspin() == 1`, it needs to be `QUDA_DEGRAND_ROSSI_GAMMA_BASIS` always.

Reply:
The exact line that fixes it: `fine_param.gammaBasis = (param.level > 0 || param.B[0].Nspin() == 1) ? QUDA_DEGRAND_ROSSI_GAMMA_BASIS : QUDA_UKQCD_GAMMA_BASIS;`